playbooks/phobos_vpn.yml

```diff
@@ -1,35 +1,101 @@
 - name: Configure Phobos VPN Tunnel (should only be run on single use slaves)
   hosts: "{{ target_hosts | default('localhost') }}"
   tasks:
-    # Use shell because python-apt isn't available so the apt module fails.
-    # I don't want to add python-apt to one of the more general slave setup
-    # scripts as it depends on python. We want to reduce the number of deps
-    # installed on general slaves to the bare minimum.
-    - name: Install apt packages
+    - name: Install ping
+      # Don't use Ansible's apt module, but instead apt-get "manually", because
+      # its dependencies, ie python-apt and Python, are probably not installed.
       shell: |
         apt-get update
-        apt-get install -y vpnc
-      register: install_packages
-      until: install_packages | success
+        apt-get install -y iputils-ping
+      register: install_ping
+      until: install_ping | success
       retries: 5
       delay: 2
       tags:
         - skip_ansible_lint
 
+    - name: Test connectivity to VPN server
+      # NOTE(coreywright): The Phobos VPN server cannot be pinged from within
+      # the Phobos environment, so use this task as a method of determining
+      # whether we are in the Phobos environment or not.
+      command: ping -c 3 {{ gateway }}
+      ignore_errors: true
+      changed_when: false
+      register: ping_vpn
 
-    - name: Template vpnc config
-      template:
-        src: vpnc.conf.j2
-        dest: /etc/vpnc/phobos.conf
+    - name: Configure and start VPN client
+      # NOTE(coreywright): If we are not in the Phobos environment, ie we can
+      # ping the Phobos VPN server, then we need to connect to it by way of
+      # VPN.
+      when: ping_vpn | succeeded
+      block:
+        # Use shell because python-apt isn't available so the apt module fails.
+        # I don't want to add python-apt to one of the more general slave setup
+        # scripts as it depends on python. We want to reduce the number of deps
+        # installed on general slaves to the bare minimum.
+        - name: Install apt packages
+          shell: |
+            apt-get update
```

Comment:
apt-get update here isn't necessary, given that it was done in the "Install ping" task

Reply:
yeah, sorry, stupid me: mindless cut-paste-minimal-edit error.